static/js/containers/DashboardPage.js

```diff
@@ -262,14 +262,14 @@ class DashboardPage extends React.Component {
 
   fetchDashboard() {
     const { dashboard, dispatch } = this.props;
-    if (dashboard.fetchStatus === undefined) {
+    if (dashboard.fetchStatus === undefined && dashboard.noSpinner === false) {
```

Comment:
If you set `fetchStatus` in all cases as described in my previous comment, you shouldn't need to check `noSpinner` at all here or below.